server/src/main/java/org/opensearch/repositories/RepositoriesService.java

```diff
@@ -763,6 +777,86 @@ public void updateRepositoriesMap(Map<String, Repository> repos) {
         }
     }
 
+    private static void ensureSystemRepositoryNotInUse(ClusterState clusterState, RepositoryMetadata repositoryMetadata) {
+        if (isRepositoryInUse(clusterState, repositoryMetadata)) {
+            throw new IllegalStateException("trying to modify or unregister repository that is currently used");
+        }
+    }
+
+    private static boolean isRepositoryInUse(ClusterState clusterState, RepositoryMetadata repositoryMetadata) {
+        // We can add more validations where we want to ensure that the system repository is not deleted if in use.
+        final Metadata idxMetadata = clusterState.getMetadata();
+        final String repository = repositoryMetadata.name();
+        for (IndexMetadata metadata : idxMetadata) {
+            if (IndexMetadata.INDEX_REMOTE_STORE_ENABLED_SETTING.get(metadata.getSettings())) {
+                String segmentRepositoryInIndex = IndexMetadata.INDEX_REMOTE_SEGMENT_STORE_REPOSITORY_SETTING.get(metadata.getSettings());
+                String translogRepositoryInIndex = IndexMetadata.INDEX_REMOTE_TRANSLOG_REPOSITORY_SETTING.get(metadata.getSettings());
+
+                if (segmentRepositoryInIndex.equals(repository) || translogRepositoryInIndex.equals(repository)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
```

Comment:
a system repository will be always be used . If not indices, cluster metadata will be uploaded there .